src/api/schema/components/sink.rs

```diff
@@ -39,9 +41,9 @@ pub struct SinksFilter {
 impl filter::CustomFilter<Sink> for SinksFilter {
     fn matches(&self, sink: &Sink) -> bool {
         filter_check!(
-            self.component_id
-                .as_ref()
-                .map(|f| f.iter().all(|f| f.filter_value(sink.get_component_id()))),
+            self.component_id.as_ref().map(|f| f
+                .iter()
+                .all(|f| f.filter_value(&sink.get_component_id().to_string()))),
```

Comment:
This is another area where I think it'd make sense to deal in `&str` rather than having to allocate a new string to filter.